Cargo.toml

```diff
@@ -41,6 +43,13 @@ tempfile = "3.14.0"
 thiserror = "2.0.3"
 # this is used by bootstrap_srv as the http server implementation.
 tiny_http = "0.12.0"
+# --- build-dependencies ---
+# The following workspace dependencies are thus-far only used in build-deps.
+# Please be careful to only include them in build dependencies or move them
+# above this section.
+# --- build-dependencies ---
+prost-build = "0.13.3"
+protobuf-src = "2.1.0"
```

Comment:
This isn't a toolset I know for Rust but does this avoid the windows build issue? https://docs.rs/prost-build/latest/prost_build/

Reply:
We are indeed using prost-build in the build.rs. It in turn requires the protoc protobuf compiler binary to be on the path, which is where the windows issue arises. I'm going to try [protoc-prebuilt](https://crates.io/crates/protoc-prebuilt) as a test to see if it works better. It should, at the very least, be faster, since it won't have to compile it : )

Reply:
That seems to have worked: https://github.com/holochain/kitsune2/pull/38/commits/bbeca4e9bbe256744a04379cec3fce6c2f1074e7 -- https://github.com/holochain/kitsune2/actions/runs/12143891872/job/33861875174?pr=38

Reply:
Nevermind, that was flaky because the runner is sometimes getting ratelimited for the binary download. Switched it to being an optional feature flag, and defaulted to having ci install protoc for use: https://github.com/holochain/kitsune2/pull/38/commits/208fec56f52462139df53d97fe1f5dd53c231193

Reply:
I definitely saw that... :) Cool okay, well this has always been my sticking point with Avro and Protobuf. They're great to use but the tooling for them can be poor.

We have to consider the downstream for this. We'll need to provide this in all our CI that builds the Kitsune API, which will be most things that interact with Holochain. I've checked it's available for Nix, and it is so that's not an issue. We'll also introduce this as a build dependency on Windows for people who compile from source.

I'd prefer to use Protobuf if it could be done without external tools everywhere, but I'm wondering the value of msgpack is just that it doesn't need external build tools to work?

Does protobuf give us a better wire format? 
I believe it gives clearer way to evolve the format
Anything else that we should consider?

Reply:
@ThetaSinner - Another option would be to build a developer generator command that would build the rust module and we could check it into source control and not require the protoc binary at compile time.

Reply:
This works: https://github.com/holochain/kitsune2/pull/38/commits/dc8ca9ba37a84aea6025c66b0bd552b64ac31dd8 we'll just have to remember to run that binary any time we update the .proto file.

Reply:
I like that! We can build some automated checks around that later too, like generate then check for a diff. Great, I'm happy then :)